src/plugin/customParseFormat/index.js

```diff
@@ -80,7 +43,55 @@ function correctHours(time) {
   }
 }
 
-function makeParser(format) {
+function makeParser(format, instance) {
+  const expressions = {
+    A: [matchUpperCaseAMPM, function (input) {
+      this.afternoon = input === 'PM'
+    }],
+    a: [matchLowerCaseAMPM, function (input) {
+      this.afternoon = input === 'pm'
+    }],
+    S: [match1, function (input) {
+      this.milliseconds = +input * 100
+    }],
+    SS: [match2, function (input) {
+      this.milliseconds = +input * 10
+    }],
+    SSS: [match3, function (input) {
+      this.milliseconds = +input
+    }],
+    s: [match1to2, addInput('seconds')],
+    ss: [match2, addInput('seconds')],
+    m: [match1to2, addInput('minutes')],
+    mm: [match2, addInput('minutes')],
+    H: [match1to2, addInput('hours')],
+    h: [match1to2, addInput('hours')],
+    HH: [match2, addInput('hours')],
+    hh: [match2, addInput('hours')],
+    D: [match1to2, addInput('day')],
+    DD: [match2, addInput('day')],
+    M: [match1to2, addInput('month')],
+    MM: [match2, addInput('month')],
+    MMM: [matchWord, function (input) {
+      const locale = instance.$locale()
+      const { months } = locale
```

Comment:
also consider `monthsShort`  in some locales

Reply:
I just started to doubt: is there a way to change locale before date parsing? According to documentation, the only way to parse is in the constructor. Am I missing something?

Reply:
why `change locale before date parsing`?